test/browser/features/steps/browser-steps.rb

```diff
@@ -112,6 +112,24 @@
     )
   end
 
+  def get_attribute_value_from_span(span, attribute, attr_type)
+    attributes = span['attributes']
+    attribute = attributes.find { |a| a['key'] == attribute }
+    value = attribute&.dig 'value', attr_type
+    attr_type == 'intValue' && value.is_a?(String) ? value.to_i : value
+  end
+
+  # Check the string attribute "referrer" is equal to document.referrer
+  referrer = get_attribute_value_from_span(span, "bugsnag.browser.page.referrer", "stringValue")
+
+  driver = Maze.driver.instance_variable_get(:@driver)
+  scenario_referrer = driver.execute_script("return document.referrer")
+
+  Maze.check.true(
+    referrer == scenario_referrer,
+    "Attribute does not match, expected #{scenario_referrer}, got #{referrer}"
```

Comment:
```suggestion
    "Referrer attribute does not match, expected #{scenario_referrer}, got #{referrer}"
```

Reply:
or is `bugsnag.browser.page.referrer` even better?